src/download.ts

```diff
@@ -277,14 +316,70 @@ export async function getFileContent(
   const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
   const url = this.getSkylinkUrl(skylinkUrl, opts);
 
-  // GET request the skylink.
+  return this.getFileContentRequest<T>(url, opts);
+}
+
+/**
+ * Gets the contents of the file at the given Handshake domain.
+ *
+ * @param this - SkynetClient
+ * @param domain - Handshake domain.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointPath="/hns"] - The relative URL path of the portal endpoint to contact.
+ * @returns - An object containing the data of the file, the content-type, metadata, and the file's skylink.
+ * @throws - Will throw if the domain does not contain a skylink.
+ */
+export async function getFileContentHns<T = unknown>(
+  this: SkynetClient,
+  domain: string,
+  customOptions?: CustomHnsDownloadOptions
+): Promise<GetFileContentResponse<T>> {
+  const opts = { ...defaultDownloadHnsOptions, ...this.customOptions, ...customOptions };
+  const url = this.getHnsUrl(domain, opts);
+
+  return this.getFileContentRequest<T>(url, opts);
+}
+
+/**
+ * Does a GET request of the skylink, returning the data property of the response.
+ *
+ * @param this - SkynetClient
+ * @param url - URL.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @returns - An object containing the data of the file, the content-type, metadata, and the file's skylink.
+ * @throws - Will throw if the request does not succeed or the response is missing data.
+ */
+export async function getFileContentRequest<T = unknown>(
+  this: SkynetClient,
+  url: string,
+  customOptions?: CustomDownloadOptions
+): Promise<GetFileContentResponse<T>> {
+  const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
+
+  // GET request the data at the skylink.
   const response = await this.executeRequest({
     ...opts,
     method: "get",
     url,
   });
 
-  return response.data;
+  if (typeof response.data === "undefined") {
+    throw new Error(
+      "Did not get 'data' in response despite a successful request. Please try again and report this issue to the devs if it persists."
+    );
+  }
+  if (typeof response.headers === "undefined") {
+    throw new Error(
+      "Did not get 'headers' in response despite a successful request. Please try again and report this issue to the devs if it persists."
+    );
+  }
+
+  // TODO: Return null instead if header not found?
```

Comment:
IMO an empty content type is acceptable, especially if we're doing it for the metadata call. It's better than `null` because the client can treat it as a string in a safe way without needing a nullish check.